src/compiler/checker.ts

```diff
@@ -4722,6 +4846,1104 @@ namespace ts {
+                        }
+                        if (p.flags & (SymbolFlags.Method | SymbolFlags.Function)) {
+                            const type = getTypeOfSymbol(p);
+                            const signatures = getSignaturesOfType(type, SignatureKind.Call);
+                            const results = [];
+                            for (const sig of signatures) {
+                                // Each overload becomes a seperate method declaration, in order
+                                const decl = signatureToSignatureDeclarationHelper(sig, methodKind, context) as MethodDeclaration;
+                                decl.name = name; // TODO: Clone
+                                if (staticFlag) {
+                                    decl.modifiers = createNodeArray(createModifiersFromModifierFlags(staticFlag));
+                                }
+                                if (p.flags & SymbolFlags.Optional) {
+                                    decl.questionToken = createToken(SyntaxKind.QuestionToken);
+                                }
+                                results.push(setTextRange(decl, sig.declaration));
+                            }
+                            return results as unknown as T[];
+                        }
+                        // The `Constructor`'s symbol isn't in the class's properties lists, obviously, since it's a signature on the static
+                        return Debug.fail(`Unhandled class member kind! ${(p as any).__debugFlags || p.flags}`);
+                    };
+                }
+
+                function serializePropertySymbolForClass(p: Symbol, isStatic: boolean, baseType: Type | undefined) {
+                    return makeSerializePropertySymbol<ClassElement>(createProperty, SyntaxKind.MethodDeclaration)(p, isStatic, baseType);
+                }
+
+                function serializePropertySymbolForInterface(p: Symbol, baseType: Type | undefined) {
+                    return makeSerializePropertySymbol<TypeElement>((_decorators, mods, name, question, type, initializer) => createPropertySignature(mods, name, question, type, initializer), SyntaxKind.MethodSignature)(p, /*isStatic*/ false, baseType);
+                }
+
+                function getDeclarationWithTypeAnnotation(symbol: Symbol) {
+                    return find(symbol.declarations, s => !!getEffectiveTypeAnnotationNode(s) && !!findAncestor(s, n => n === enclosingDeclaration));
+                }
+
+                /**
+                 * Unlike `typeToTypeNodeHelper`, this handles setting up the `AllowUniqueESSymbolType` flag
+                 * so a `unique symbol` is returned when appropriate for the input symbol, rather than `typeof sym`
+                 */
+                function serializeTypeForDeclaration(type: Type, symbol: Symbol) {
+                    const declWithExistingAnnotation = getDeclarationWithTypeAnnotation(symbol);
+                    if (declWithExistingAnnotation
```

Comment:
it would be kind of nice to have serializeSymbol[Worker] at the top of the serialize* functions.